### ISSUE TITLE
Reading from S3 storage can cause Prefetch is valid in readUntilPosition

### ISSUE THREAD

**Issue by minaevmike** [ClickHouse/ClickHouse#38628]
**Describe what's wrong**
I have a `ReplicatedMergeTree`. It stores data both on local disk and remote S3 storage. In some cases, the select queries failed with the following error.
```
Code: 49. DB::Exception: Received from host:9000. DB::Exception: Received from host:9000. DB::Exception: Prefetch is valid in readUntilPosition: (while reading column order_number): (while reading from part /ch_data0/clickhouse/disks/s3/store/129/129717bc-0d26-4af7-9297-17bc0d266af7/20220301_0_91996_27/ from mark 9930 with max_rows_to_read = 1601): While executing MergeTreeThread.
``` 
This only happens when the result set is pretty huge(in my case over 300k rows). I've found similar issue(https://github.com/ClickHouse/ClickHouse/pull/33046), but in my case `order_number` has `String` type.

Also, if I set `remote_filesystem_read_prefetch=0` everything works fine.

***Version***
```
SELECT version()

┌─version()─┐
│ 22.3.2.1  │
└───────────┘
```

**Comment by minaevmike** [comment]
One important note. This only happens when I have a partition where 1 part located on local disk and other part stored at s3 storage.

**Comment by alexey-milovidov** [comment]
Does it reproduce on the latest version from the master?
```
curl https://clickhouse.com/ | sh
```

**Comment by kssenii** [comment]
This is fixed in 22.6 (https://github.com/ClickHouse/ClickHouse/pull/36763)
